app/services/event_generator/composite_generator.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+class EventGenerator
+  class CompositeGenerator
+    attr_reader :generators
+
+    def initialize(generators)
+      @generators = generators
+    end
+
+    def record_created(record)
+      delegate_to_generator(__method__, record)
+    end
+
+    def record_deleted(record)
+      delegate_to_generator(__method__, record)
+    end
+
+    def record_updated(record)
+      delegate_to_generator(__method__, record)
+    end
+
+    private
+
+      # Send method with record argument to first valid generator
+      def delegate_to_generator(method_name, record)
+        generators.each do |generator|
```

Comment:
Can probably just do `generators.find{|generator| generator.valid?(record)}.send(method_name, record)`

Reply:
I think we still need to check if the generator responds to the method in addition to validity. Alternatively, each generator needs to respond to all of the composite generator methods.  For example, the Geoblacklight generator only responds to `record_updated` and `record_deleted` at the moment. We would have to add an empty method definition:

```
def record_created; end
```

Thoughts? I could go either way.

Reply:
I'm personally of the opinion that we should have the empty method definition. I'm surprised they pass the shared spec otherwise - I would expect `record_created` to be part of the interface.